src/core/algorithms/dd/split/split.cpp

```diff
@@ -249,45 +290,59 @@ unsigned Split::RemoveTransitiveDDs() {
 void Split::CalculateIndexSearchSpaces() {
     std::vector<DFConstraint> new_min_max_dif;
     std::vector<std::vector<std::vector<double>>> new_distances;
+    std::vector<DistancePositionListIndex> new_plis;
     new_min_max_dif.reserve(num_columns_);
     new_distances.reserve(num_columns_);
+    new_plis.reserve(num_columns_);
     for (model::ColumnIndex index = 0; index < num_columns_; index++) {
         std::vector<DFConstraint> cur_index_search_space = IndexSearchSpace(index);
         if (!cur_index_search_space.empty()) {
             index_search_spaces_.push_back(std::move(cur_index_search_space));
             non_empty_cols_.push_back(index);
             new_min_max_dif.push_back(min_max_dif_[index]);
             new_distances.push_back(std::move(distances_[index]));
+            new_plis.push_back(std::move(plis_[index]));
         }
     }
     num_columns_ = non_empty_cols_.size();
     min_max_dif_ = std::move(new_min_max_dif);
     distances_ = std::move(new_distances);
+    plis_ = std::move(new_plis);
+}
+
+void Split::CalculateTuplePairs() {
+    std::size_t const df_search_space_num = std::accumulate(
+            index_search_spaces_.begin(), index_search_spaces_.end(), 0,
+            [](std::size_t acc, auto const& search_space) { return acc + search_space.size(); });
+
+    std::unordered_set<boost::dynamic_bitset<>> tuple_pair_set;
+    for (std::size_t first_index = 0; first_index < num_rows_; first_index++) {
+        for (std::size_t second_index = first_index + 1; second_index < num_rows_; second_index++) {
+            boost::dynamic_bitset<> pair_bitset(df_search_space_num);
+            std::size_t df_index = 0;
+
+            for (model::ColumnIndex column_index = 0; column_index < num_columns_; column_index++) {
+                for (auto const& df_constraint : index_search_spaces_[column_index]) {
+                    if (CheckDFConstraint(df_constraint, column_index,
+                                          {first_index, second_index})) {
+                        pair_bitset.set(df_index);
+                    }
```

Comment:
If you want, you can change it like so:
```suggestion
    pair_bitset.set(df_index, CheckDFConstraint(df_constraint, column_index, {first_index, second_index}));
```
but do as you like